src/app/component/projects/pages/details/details.component.html

```diff
@@ -56,18 +57,46 @@ <h5>
         </li>
       </ul>
     </div>
-    <div class="col-sm-6"></div>
+
+    <!-- Right aligned map -->
+    <div class="col-sm-6">
+      <ng-container *ngIf="sites && sites.length > 0; else placeholderMap">
+        <agm-map [latitude]="avgLat" [longitude]="avgLong">
+          <ng-container *ngFor="let site of sites">
+            <agm-marker
+              *ngIf="site.customLongitude && site.customLatitude"
+              [latitude]="site.customLatitude"
+              [longitude]="site.customLongitude"
+            >
+              <agm-snazzy-info-window
+                [isOpen]="true"
+                [padding]="'10px 25px 10px 10px'"
+                [borderRadius]="'10px'"
+              >
+                <ng-template>{{ site.name }}</ng-template>
+              </agm-snazzy-info-window>
+            </agm-marker>
+          </ng-container>
+        </agm-map>
+      </ng-container>
+      <ng-template #placeholderMap>
+        <div class="map-placeholder"><span>No locations specified</span></div>
+      </ng-template>
+    </div>
   </div>
 </ng-container>
 
+<!-- Unauthorized Access Display -->
 <ng-container *ngIf="state === 'unauthorized'">
   <app-unauthorized></app-unauthorized>
 </ng-container>
 
+<!-- Project Not Found Display -->
 <ng-container *ngIf="state === 'notFound'">
   <app-item-not-found></app-item-not-found>
 </ng-container>
 
```

Comment:
It seems strange these are part of the details view. Do they have to be specified for every view? Is there a more general way of doing this?

Reply:
I would almost prefer a more general way of performing this. Perhaps a component which takes a state variable as an input?

Reply:
So when errors like this happen, the routing system should render an error component in place of the component that was meant to be rendered.

E.g. 404 on projects/123 should render <not-found /> in place of <project-details />

This should be a separate issue, let's not fix it here. Create an issue, please.

Reply:
Issue created here #30.